pennylane/ops/qubit.py

```diff
@@ -447,6 +454,84 @@ def decomposition(wires):
         return decomp_ops
 
 
+class CU3(Operation):
+    r"""CU3(theta, phi, lambda, wires)
+    The controlled arbitrary single qubit unitary operator.
+
+    .. math::
+
+        CU3(\theta, \phi, \lambda)\ =
+        |0\rangle\langle 0| \otimes I +
+        |1\rangle\langle 1| \otimes U3(\theta,\phi,\lambda) =
+        \begin{bmatrix}
+            1 & 0 & 0 & 0 \\
+            0 & 1 & 0 & 0\\
+            0 & 0 & \cos(\theta/2) & -\exp(i \lambda)\sin(\theta/2)\\
+            0 & 0 & \exp(i \phi)\sin(\theta/2) & \exp(i (\phi + \lambda))\cos(\theta/2)
+        \end{bmatrix}.
+
+    .. note:: The first wire provided corresponds to the **control qubit**.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 3
+
+    Args:
+        theta (float): polar angle :math:`\theta`
+        phi (float): azimuthal angle :math:`\phi`
+        lambda (float): quantum phase :math:`\lambda`
+        wires (Sequence[int] or int): the subsystem the gate acts on
+    """
+
+    num_params = 3
+    num_wires = 2
+    par_domain = "R"
+    grad_method = "F"
```

Comment:
How come you have chosen `grad_method="F"` here?

Reply:
Ah, I wasn't able to get it to work with the analytic differentiation option (I see now that it might be because the decomposition has addition of variables), and I thought using the finite difference method would be safe since it's the default method anyways? Please let me know if I should change it!